### ISSUE TITLE
Could not find an IRouter associated with the ActionContext

### ISSUE THREAD

**Issue by RedXLion** [fpanaccia/Wkhtmltopdf.NetCore#19]
Hi!

Just testing your tool in VS2019 ASP.Net Core 3.1 project.
Stuck in this message "Could not find an IRouter associated with the ActionContext..."
Works only with hard coded html string without any model added.
Any help is kindly appreciated!

Thank you in advance 
RL

**Comment by fpanaccia** [comment]
Hi!, what version of the tool are you using? i just tested using my example and worked

![image](https://user-images.githubusercontent.com/5436904/78056813-184b6080-735c-11ea-85b3-e4249e10a641.png)

could you upload the project or at least the basic structure?

**Comment by RedXLion** [comment]
Thank you for the quick response!
That's the thing. Your example is working! But when I try to add it to the new project  as is ,I receive the error. I use files from the example.

**Comment by RedXLion** [comment]
[https://github.com/RedXLion/Test](url)

**Comment by fpanaccia** [comment]
There is no csproj in that repo

**Comment by RedXLion** [comment]
Ready

**Comment by fpanaccia** [comment]
can i make some changes and make a pr to your test repo?

**Comment by RedXLion** [comment]
sure

**Comment by fpanaccia** [comment]
There, working

![image](https://user-images.githubusercontent.com/5436904/78061941-56e51900-7364-11ea-9333-8e51eae4a077.png)

**Comment by fpanaccia** [comment]
[https://github.com/RedXLion/Test/pull/1](url)

**Comment by RedXLion** [comment]
Could you please, explain a bit what exactly gone wrong?
Thank you very much!

**Comment by fpanaccia** [comment]
The problem was with AddControllersWithViews, in your project, you didnt have any views, also the tool of wkthmltopdf was missing

So, in another words, i have added the binary of wkthmltopdf, added the view and fixed up a little bit the startup.cs

**Comment by fpanaccia** [comment]
If you replace in the PR, these lines

`services.AddWkhtmltopdf();
 services.AddControllers();`

with these lines
`services.AddWkhtmltopdf();
services.AddControllersWithViews();`

Now it will work too because there is a view in the project.

**Comment by RedXLion** [comment]
Thanks a lot for your time!
Very appreciate that.
Keep you and family in good wells!

**Comment by fpanaccia** [comment]
Your welcome!

**Comment by RedXLion** [comment]
Dear f.

Really sorry to bother you again.
I've created a new project with views and tried to implement the labrary, now with respect to all your specifications. So, it didn't worked.
Until I've removed the _layout.cshtml reference from the _ViewStart.cshtml.
Do I miss something?

Best regards,
RL

On Mar 31, 2020 22:18, fpanaccia <notifications@github.com> wrote:

Closed #19<https://github.com/fpanaccia/Wkhtmltopdf.NetCore/issues/19>.
